docs/tutorials/self-hosting.md

```diff
@@ -0,0 +1,98 @@
+---
+sidebar_position: 25
+title: Self-Hosting
+description: Learn how to self-host DBOS Transact applications
+---
+
+You can run DBOS applications anywhere with `npx dbos start` as long as they have a Postgres database to connect to.
+This guide describes tools that make it easy to self-host DBOS applications in production.
+
+## Admin API
+
+DBOS applications expose an admin API, fixed at one above the main DBOS application port (the main port defaults to port 3000, so the admin API defaults to port 3001).
+It provides the following endpoints:
+
+### Health Check
+
+- **Endpoint**: `/dbos-healthz`
+- **HTTP Method**: GET
+- **Description**: Performs a health check on the application.
+- **Response**:
+  - **Status Code**: 200 OK if the system is healthy; otherwise, appropriate error codes.
+
+### Workflow Recovery
+
+
+- **Endpoint**: `/dbos-workflow-recovery`
+- **Method**: POST
+- **Description**: Recovers all pending workflows associated with input [executor IDs](#managing-workflow-recovery). Following our [reliability guarantees](./workflow-tutorial.md#reliability-guarantees), all workflows will resume from where they left off. Returns the UUIDs of all workflows recovered.
+- **Request Body Format**: JSON list of executors whose pending workflows to recover.
+  - **Example**:
+    ```json
+      ["executor-id-1", "executor-id-2", "..."]
+- **Response**:
+  - **Status Code**: 200 OK on successful recovery initiation; otherwise, appropriate error codes.
+  - **Body Format**: JSON list of UUIDs representing the workflows that were successfully queued for recovery.
+  - **Example**:
+    ```json
+    ["workflow-uuid-1", "workflow-uuid-2", "..."]
+    ```
```

Comment:
So what happens if only part of the workflows are successfully recovered? 500 and a list of successfully recovered workflows in the body?